quickwit-search/src/export/root.rs

```diff
@@ -0,0 +1,194 @@
+// Quickwit
+//  Copyright (C) 2021 Quickwit Inc.
+//
+//  Quickwit is offered under the AGPL v3.0 and as commercial software.
+//  For commercial licensing, contact us at hello@quickwit.io.
+//
+//  AGPL:
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Affero General Public License as
+//  published by the Free Software Foundation, either version 3 of the
+//  License, or (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Affero General Public License for more details.
+//
+//  You should have received a copy of the GNU Affero General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::io::Write;
+use std::sync::Arc;
+
+use futures::StreamExt;
+use quickwit_proto::ExportRequest;
+use quickwit_proto::LeafExportRequest;
+use tracing::*;
+
+use quickwit_metastore::{Metastore, SplitMetadata};
+use quickwit_proto::SearchRequest;
+
+use crate::client_pool::Job;
+use crate::error::parse_grpc_error;
+use crate::list_relevant_splits;
+use crate::root::job_for_splits;
+use crate::root::NodeSearchError;
+use crate::ClientPool;
+use crate::SearchClientPool;
+use crate::SearchError;
+
+/// Perform a distributed export.
+pub async fn root_export(
+    export_request: &ExportRequest,
+    metastore: &dyn Metastore,
+    client_pool: &Arc<SearchClientPool>,
+) -> Result<Vec<u8>, SearchError> {
```

Comment:
Do we have to return a `Vec<u8>` ? 
we could return a `Vec<Vec<u8>>` and save on the extra allocation line `103`